citesphere/src/main/java/edu/asu/diging/citesphere/core/repository/oauth/DbAccessTokenRepository.java

```diff
@@ -18,11 +17,16 @@
     List<DbAccessToken> findByClientId(String clientId);
  
     List<DbAccessToken> findByClientIdAndUsername(String clientId, String username);
+    
+    List<DbAccessToken> findByUsername(String username);
  
     Optional<DbAccessToken> findByTokenId(String tokenId);
  
     Optional<DbAccessToken> findByRefreshToken(String refreshToken);
  
     Optional<DbAccessToken> findByAuthenticationId(String authenticationId);
+   
+    @Transactional
```

Comment:
this should not be necessary

Reply:
Without this annotation it is giving following error:

No EntityManager with actual transaction available for current thread - cannot reliably process 'remove' call; nested exception is javax.persistence.TransactionRequiredException: No EntityManager with actual transaction available for current thread - cannot reliably process 'remove' call] with root cause

Reply:
the transaction should be on the manager class that's calling the repo